### PR TITLE
Added handling for errors while building the packet.

### PR DESCRIPTION
**Pull request by come-maiz**

### CODE REVIEW

File: subunit.go [hunk 1]
```diff
@@ -127,10 +132,12 @@ func (e *Event) write(writer io.Writer) error {
 	var bTemp bytes.Buffer
 	bTemp.WriteByte(signature)
 	bTemp.Write(<-flagsChan)
-	bTemp.Write(<-timestampChan)
-	bTemp.Write(<-idChan)
-	bTemp.Write(<-mimeChan)
-	bTemp.Write(<-fileContentChan)
+	for _, part := range []packetPart{<-timestampChan, <-idChan, <-mimeChan, <-fileContentChan}{
```
Comment (come-maiz) [inline subunit.go:135]: @fgimenez I have a question. Do you know if this blocks until the last element of the slice is created, or it starts consuming them (and potentially blocking) one by one?

**Comment by fgimenez** [comment]
Let's check [1]. The first element in the array waits 1 second and the last 2 seconds, but the loop only begins when all of them, including the slowest, have sent a value.

[1] http://play.golang.org/p/asWbl-dfnT

**Comment by come-maiz** [comment]
@fgimenez I think I can make a channel that waits for the other channels in order. Are you ok with the rest of the branch? As I still don't know how to make prerequisites, I prefer to land things first.

**Comment by fgimenez** [comment]
mmm I think that what you have is enough, the for waits for all the channels to finish and then iterates in order, do you need something different? The example wasn't clear [1] 

I'm ok with the rest of the branch as well :)

[1] http://play.golang.org/p/t-LrGDEu4z

**Comment by come-maiz** [comment]
The example was clear, I forgot to thank you about it :) I'm just using this to learn about channels. So we are not saving any real time, but I'd like the loop to start as soon as the first channel is ready. I'll merge this, and keep playing when I have some free time.

**Pull request merged**